### PR TITLE
add ctx propagation on all lib functions

### PR DESCRIPTION
**Pull request by neymarjimoh**
Enhancing the lib functions to include context for increased user controls would be valuable. This would give the end developer utilizing this library more control from within their applications.

```
payload := fastotp.GenerateOTPPayload{
    Delivery: OtpDelivery{
      "email": "example@example.com",
    },
		Identifier:  "user123",
		TokenLength: 6,
		Type:        "numeric",
		Validity:    120,
	}

// Generate OTP
otp, err := client.GenerateOTP(ctx, payload)
// ... I can cancel or set timeout based on my business rules
```

### CODE REVIEW

File: lib/http.go [hunk 1]
```diff
@@ -11,13 +12,15 @@ import (
...
+	Ctx     context.Context
 }
 
 // NewAPIClient creates a new instance of APIClient.
-func NewAPIClient(baseURL, apiKey string) *APIClient {
+func NewAPIClient(baseURL, apiKey string, ctx context.Context) *APIClient {
 	return &APIClient{
 		BaseURL: baseURL,
 		APIKey:  apiKey,
+		Ctx:     ctx,
```
Comment (LordRahl90) [inline lib/http.go]: there's no point exporting these values, as you don't want them changed/accessed directly anyways

**Review by LordRahl90 — CHANGES_REQUESTED** [review]
Great work, 
However, there's no need to expose the values in the `APIClient`.

**Comment by neymarjimoh** [comment]
> Great work, However, there's no need to expose the values in the `APIClient`.

Thanks for the feedback. I agree that exposing `BaseURL` and `APIKey` directly may not be ideal. I just added the Ctx key for managing user's context that's why and noticed your PR handled that. Will modify now

File: go.mod [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module github.com/CeoFred/fast-otp
 
-go 1.21.2
+go 1.23
```
Comment (CeoFred) [inline go.mod]: Go version is `1.21.2` work with that.

**Review by CeoFred — CHANGES_REQUESTED** [review]

**Review by LordRahl90 — APPROVED** [review]

**Comment by LordRahl90** [comment]
> > Great work, However, there's no need to expose the values in the `APIClient`.
> 
> Thanks for the feedback. I agree that exposing `BaseURL` and `APIKey` directly may not be ideal. I just added the Ctx key for managing user's context that's why and noticed your PR handled that. Will modify now

Great work boss 🙇‍♂️

**Comment by neymarjimoh** [comment]
> > > Great work, However, there's no need to expose the values in the `APIClient`.
> > 
> > 
> > Thanks for the feedback. I agree that exposing `BaseURL` and `APIKey` directly may not be ideal. I just added the Ctx key for managing user's context that's why and noticed your PR handled that. Will modify now
> 
> Great work boss 🙇‍♂️

Thanks boss

**Pull request merged**